source/context.cpp

```diff
@@ -429,7 +430,14 @@ void Context::generate(Config const &config)
 
 		if( i < binders.size() ) --i;
 
-		code = generate_include_directives(includes) + fmt::format(module_header, config.includes_code()) + prefix_code + "void " + function_name + module_function_suffix + "\n{\n" + code + "}\n";
+		string const custom_shared = Config::get().custom_shared();
+		bool use_custom_shared = Config::get().use_custom_shared();
+
+		string custom_shared_declare = "";
+		if( use_custom_shared )
+			custom_shared_declare = "PYBIND11_DECLARE_HOLDER_TYPE(T, "+custom_shared+"<T>)";
```

Comment:
Should we instead replace `PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>)` line with our custom shared line?